devito/archinfo.py

```diff
@@ -60,18 +62,36 @@ def get_cpu_brand():
             except:
                 pass
 
-            return None
+            try:
+                return cpuinfo.get_cpu_info().get('raw_arch_string')
+            except:
+                pass
 
         cpu_info['flags'] = get_cpu_flags()
         cpu_info['brand'] = get_cpu_brand()
 
-    if not all(i in cpu_info for i in ('flags', 'brand')):
-        # Fallback
-        ci = cpuinfo.get_cpu_info()
-        cpu_info['flags'] = ci.get('flags')
-        cpu_info['brand'] = ci.get('brand')
+    try:
+        if 'flags' not in cpu_info:
+            # Fallback
+            ci = cpuinfo.get_cpu_info()
+            cpu_info['flags'] = ci.get('flags')
```

Comment:
so going back to my original comment. What I was **actually** suggesting was to move the **content** of the try-except in place of current line 48 -- instead of returning `None`, we return the `cpuinfo.get_cpu_info().get('flags')` ?

Reply:
again, I did that the previous days and was failing on OSX